src/nnet2/nnet-component.cc

```diff
@@ -3933,7 +3933,7 @@ void Convolutional1dComponent::Propagate(const ChunkInfo &in_info,
   }
   
   // apply all filters
-  AddMatMatBatched(1.0f, tgt_batch, patch_batch, kNoTrans, filter_params_batch,
+  AddMatMatBatched<BaseFloat>(1.0f, tgt_batch, patch_batch, kNoTrans, filter_params_batch,
```

Comment:
Since we're at it, could you please replace the 1.0f in these AddMatMatBatched calls with 1.0?
1.0f seems to have been an attempt to call the <float> version of the template, but isn't needed and is confusing.